test/e2e/network/dual_stack.go

```diff
@@ -0,0 +1,308 @@
+			gomega.Expect(isIPv6CIDR(node.Spec.PodCIDRs[1])).Should(gomega.BeTrue())
+		}
+	})
+
+	ginkgo.It("should create pod, add ipv6 and ipv4 ip to pod ips", func() {
+		podName := "pod-dualstack-ips"
+
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"test": "dualstack-pod-ips"},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "dualstack-pod-ips",
+						Image: networkUtilImage,
+					},
+				},
+			},
+		}
+
+		ginkgo.By("submitting the pod to kubernetes")
+		podClient.CreateSync(pod)
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+
+		p, err := podClient.Get(pod.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err, "Failed to get pod %q", pod.Name)
+
+		gomega.Expect(p.Status.PodIP).ShouldNot(gomega.BeEquivalentTo(""))
+		gomega.Expect(p.Status.PodIPs).ShouldNot(gomega.BeNil())
+
+		// validate there are 2 ips in podIPs
+		gomega.Expect(len(p.Status.PodIPs)).Should(gomega.BeNumerically("==", 2))
+		// validate first ip in PodIPs is same as PodIP
+		gomega.Expect(p.Status.PodIPs[0].IP).Should(gomega.BeEquivalentTo(p.Status.PodIP))
+		// validate second ip in PodIPs is ipv6
+		gomega.Expect(isIPv6(p.Status.PodIPs[1].IP)).Should(gomega.BeTrue())
+
+		ginkgo.By("deleting the pod")
+		err = podClient.Delete(pod.Name, metav1.NewDeleteOptions(30))
+		framework.ExpectNoError(err, "failed to delete pod")
+	})
+
+	// takes close to 140s to complete, so doesn't need to be marked [SLOW]
+	ginkgo.It("should be able to reach pod on ipv4 and ipv6 ip", func() {
+		nginxDeploymentName := "dualstack-server-nginx"
+		networkUtilsDeploymentName := "dualstack-client-utils"
+
+		// get all schedulable nodes to determine the number of replicas for pods
+		// this is to ensure connectivity from all nodes on cluster
+		nodeList := framework.GetReadySchedulableNodesOrDie(cs)
+		gomega.Expect(nodeList).NotTo(gomega.BeNil())
+
+		if len(nodeList.Items) < 1 {
+			framework.Failf("Expect at least 1 node, got %v", len(nodeList.Items))
+		}
+
+		replicas := int32(len(nodeList.Items))
+
+		// to ensure all the pods land on different nodes and we can thereby
+		// validate connectivity across all nodes.
+		serverDeploymentSpec := &apps.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   nginxDeploymentName,
+				Labels: map[string]string{"test": "dualstack-test-server"},
+			},
+			Spec: apps.DeploymentSpec{
+				Replicas: &replicas,
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"test": "dualstack-test-server"}},
+				Strategy: apps.DeploymentStrategy{
+					Type: apps.RollingUpdateDeploymentStrategyType,
+				},
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"test": "dualstack-test-server"},
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:  "dualstack-test-server",
+								Image: nginxImageWithIPv6,
```

Comment:
we can use this `Image: imageutils.GetE2EImage(imageutils.TestWebserver),` , it's a web server listening on both ipv4 and ipv6 and is very small https://github.com/kubernetes/kubernetes/pull/75246
Also, IIRC current nginx official docker images are dual stack by default, we can use
`Image: imageutils.GetE2EImage(imageutils.Nginx)` if we need nginx specifically

Reply:
The current nginx image listens only on ipv4, which is why I built an image with server listening on ipv6. I'll try out the test web server.

Reply:
you are absolutely right, but you can use `Image: Imageutils.GetE2EImage(imageutils.TestWebserver)` 
I'm trying to add another webserver image with dual stack to fix other tests, I'll keep you posted